templates/repo/view_list.tmpl

```diff
@@ -9,7 +9,7 @@
 					<img class="ui avatar image img-12" src="{{AvatarLink .LatestCommit.Author.Email}}" />
 					<strong>{{.LatestCommit.Author.Name}}</strong>
 				{{end}}
-				<a rel="nofollow" class="ui sha label" href="{{.RepoLink}}/commit/{{.LatestCommit.ID}}" rel="nofollow">{{ShortSHA1 .LatestCommit.ID.String}}</a>
+				<a rel="nofollow" class="ui sha label" href="{{.RepoLink}}/tree/{{.LatestCommit.ID}}" rel="nofollow">{{ShortSHA1 .LatestCommit.ID.String}}</a>
```

Comment:
I don't think `.../tree/...` is supported with our current version of gogs. I think the appropriate change would be `.../src/...` to achieve the required result.

Reply:
Also this change affects general files and is not submodule specific as the PR title describes. Submodule specific code starts from line 30 onwards.